projects/sasil-web/src/components/molelcules/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import * as styles from './DropdownMenu.style';
+
+export interface DropdownMenuProps {
+  /** 메뉴 노출 정보 객체 */
+  menuDisplayInfo: { display: boolean; top: number; left: number };
+  /** 메뉴 내부에 들어가는 HTML */
+  children?: React.ReactNode;
+  /** 메뉴 노출 컨트롤 함수 */
+  onMenuDisplayChange?: (top?: number, left?: number) => void;
+}
+
+/** 드롭다운 메뉴를 만들어주는 컴포넌트 */
+const DropdownMenu = ({
+  menuDisplayInfo,
+  onMenuDisplayChange,
+  children,
+}: DropdownMenuProps) => (
+  <styles.MenuWrap top={menuDisplayInfo.top} left={menuDisplayInfo.left}>
+    {menuDisplayInfo.display && (
+      <>
+        <styles.Menu>{children}</styles.Menu>
+        <styles.FixedBackground onClick={() => onMenuDisplayChange?.()} />
```

Comment:
optional chaining이 이렇게도 쓰일 수 있구나..! 하나 배워갑니다:)
그런데 `onMenuDisplayChange` 함수가 `undefined`가 될 수 있는 상황을 만들어놓은 이유는 재사용성 때문인가여?

Reply:
🥝 제 생각이 맞다면 이렇게도 쓸 수 있을 것 같아요!
```suggestion
        <styles.FixedBackground onClick={onMenuDisplayChange} />
```

Reply:
@aube-dev 아 내가 그렇게 못했던 이유를 보니까, 
`onMenuDisplayChange` 가 타입이 `(top?:number,left?number)=> void` 이라,  top,left를 인자로 전달해주면 dropdown menu 위치가 전달한 값으로 변경되고, 인자를 전달하지 않으면 그냥 위치를 0으로 설정해주는 함수라 저렇게 하면 에러가 떴었던 것 같아여! 
(말하다보니 너무 복잡하게 짠 것 같기도..)
![image](https://user-images.githubusercontent.com/67703882/181404766-02ab6a28-eb83-48ec-bf99-1dd856d54d7a.png)

Reply:
@kanghyun98 뭔가 추후에 드롭다운 메뉴 관련한 컴포넌트에 `onMenuDisplayChange` 함수가 없어도 될 상황?이 올 수도 있다 생각하여.. 일단 ?를 붙였던 것 같습니다,,!

Reply:
아하 그런 상황이면 저게 맞긴 하겠다! 굿굿